analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -0,0 +1,194 @@
+import org.ossreviewtoolkit.spdx.toSpdx
+
+private const val DEFAULT_SCOPE_NAME = "default"
+
+/**
+ * A "fake" package manager implementation that uses SPDX documents as "marker" files to declare projects and describe
+ * packages. See https://github.com/spdx/spdx-spec/issues/439 for details.
+ */
+class SpdxDocumentFile(
+    managerName: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(managerName, analysisRoot, analyzerConfig, repoConfig) {
+    class Factory : AbstractPackageManagerFactory<SpdxDocumentFile>("SpdxDocumentFile") {
+        override val globsForDefinitionFiles = listOf("*.spdx.yml", "*.spdx.yaml", "*.spdx.json")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = SpdxDocumentFile(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    private fun SpdxPackage.toIdentifier() =
+        Identifier(
+            type = managerName,
+            namespace = originator.extractOrganization().orEmpty(),
+            name = name,
+            version = versionInfo
+        )
+
+    private fun String.extractOrganization() =
+        lineSequence().mapNotNull { line ->
+            line.removePrefix("Organization: ").takeIf { it != line }
+        }.firstOrNull()
+
+    /**
+     * Return the dependencies of [pkg] defined in [doc] of the [SpdxRelationship.Type.DEPENDENCY_OF] type.
+     */
+    private fun getDependencies(pkg: SpdxPackage, doc: SpdxDocument): SortedSet<PackageReference> =
+        getDependencies(pkg, doc, SpdxRelationship.Type.DEPENDENCY_OF) { target ->
+            val dependency = doc.packages.single { it.spdxId == target }
+            PackageReference(
+                id = dependency.toIdentifier(),
+                dependencies = getDependencies(dependency, doc)
+            )
+        }
+
+    /**
+     * Return the dependencies of [pkg] defined in [doc] of the given [dependencyOfRelation] type. Optionally, the
+     * [SpdxRelationship.Type.DEPENDS_ON] type is handled by [dependsOnCase].
+     */
+    private fun getDependencies(
+        pkg: SpdxPackage,
+        doc: SpdxDocument,
+        dependencyOfRelation: SpdxRelationship.Type,
+        dependsOnCase: (String) -> PackageReference? = { null }
+    ): SortedSet<PackageReference> =
+        doc.relationships.mapNotNullTo(sortedSetOf()) { (source, target, relation) ->
+            when {
+                // Dependencies can either be defined on the target...
+                target == pkg.spdxId && relation == dependencyOfRelation -> {
+                    val dependency = doc.packages.single { it.spdxId == source }
+                    PackageReference(
+                        id = dependency.toIdentifier(),
+                        dependencies = getDependencies(dependency, doc, dependencyOfRelation, dependsOnCase)
```

Comment:
Is it correct to pass `dependencyOfRelation` to the recursive call? I wonder if `SpdxRelation.DEPENDENCY_OF` should be passed instead..

Reply:
I agree that passing `DEPENDENCY_OF` makes more sense, will fix.

Reply:
This could as well be addressed in a follow-up PR.

Reply:
It's already addressed...